Adding supporting python scripts for setup configuration and output read and diagnostics

rachmeister opened the pull request:
Added a directory with python scripts and notebooks containing expanded background profile setup and for parsing and performing basic diagnostics on simulation output.

Commit:
Adding python scripts and notebooks with expanded background profile setup and to parse and perform basic diagnostics on output

iangrooms commented on setup_and_diagnostics/.ipynb_checkpoints/Basic output read and diagnostics-checkpoint.ipynb:
Please update the .gitignore so that we don't track the Jupyter notebook checkpoints, then delete this file.

rachmeister replied:
Done in update

iangrooms commented on setup_and_diagnostics/.ipynb_checkpoints/Generate simulation setup and profiles-checkpoint.ipynb:
Please update the .gitignore so that we don't track the Jupyter notebook checkpoints, then delete this file.

rachmeister replied:
Done in update

iangrooms commented on setup_and_diagnostics/Generate simulation setup and profiles.ipynb:
In cell 4, please repace 'constant grid' by 'equispaced grid'. 

Also, it would be nice to add a full Chebyshev grid, but not necessary.

Please change the names of the Charney and Ocean-Charney grids. What is currently "Charney" should be "Ocean Charney" while what is currently "Ocean Charney" is a combination of the first baroclinic mode and the ocean-Charney profile. So I think you should re-name Charney to ocean-charney and then pick a different name for what is currently called ocean-charney. It would also be good to change the hard-coded values 0.22 and 0.08 to optional arguments.

rachmeister replied:
Naming has been adjusted, fractional values for the combined background profile are now parameters, and I've streamlined using different spaced grids with the alpha coordinates including a full Chebyshev option.

iangrooms commented:
Thanks for this PR @rachmeister! This will help make the code more user-friendly. I'll continue to evaluate and request changes, and hopefully we can get this merged soon.

Commit:
Remove and ignore notebook checkpoints

Add .gitignore

Commit:
Cleaning up naming and notation per PR and adding/streamlining analytic alpha grid generation with full Chebyshev option now included

iangrooms commented:
Thanks for these changes. Can you change the default parameters.f90 file so that nz and Htot match your notebooks? That way it will be less confusing for a first-timer.

I ran a test with this setup that takes 11 steps. When I tried using the 'Basic output read and diagnostics' notebook it plots the following for time steps
<img width="861" height="679" alt="BORAD" src="https://github.com/user-attachments/assets/f5747e40-d2a3-44a7-8d16-26dfbe7ae874" />
and the KE, APE, and total energy plots are blank. (The horizontal cross sections plot correctly, as does the horizontal KE spectrum.) Is this the expected behavior?

Commit:
restoring missing file in compile, fixing output log scrape for updated format, and tweaking input to match QGN parameter file

rachmeister commented:
> Thanks for these changes. Can you change the default parameters.f90 file so that nz and Htot match your notebooks? That way it will be less confusing for a first-timer.
> 
> I ran a test with this setup that takes 11 steps. When I tried using the 'Basic output read and diagnostics' notebook it plots the following for time steps <img alt="BORAD" width="861" height="679" src="https://private-user-images.githubusercontent.com/25990205/474584658-f5747e40-d2a3-44a7-8d16-26dfbe7ae874.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1MDc2OTYsIm5iZiI6MTc1NDUwNzM5NiwicGF0aCI6Ii8yNTk5MDIwNS80NzQ1ODQ2NTgtZjU3NDdlNDAtZDJhMy00NGE3LThkMTYtMjZkZmJlN2FlODc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA2VDE5MDk1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2YmYwNTM1MGQ3YTk0YjQyMTM5NjZmMTMzNmJjNjcxOWZmZTc0YTE3YzYyNGQ2M2QwZjFmOWExYzJlNmY2NjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wueAQboVGSegTYf-ODIq5lnZ17YkKJiqmCnjb47OEAo"> and the KE, APE, and total energy plots are blank. (The horizontal cross sections plot correctly, as does the horizontal KE spectrum.) Is this the expected behavior?

Thanks for the review and comments @iangrooms! Glad to get any wrinkles ironed out and the notebooks into a user-friendly, streamlined state.

I’ve updated the notebook to reflect the parameters file so it’s all consistent for a first-time test run. Happy to adjust if there’s a different,preferred configuration for the default.

The scraping from the log file was not expected behavior and has been updated to correctly handle the slightly different format in the log. I ran through a small test from a clean clone and found that the “mkl_dfti.f90” file was missing and caused the compile to fail  — I’ve restored it from my working copy. The energy diagnostics are computed only from the full output files and weren’t showing up in the plot with just a single point in time. I’ve updated the notebook to make this usage clearer, use plot markers so single points will show up, and include a few timesteps in the default.

iangrooms commented on mkl_dfti.f90:
Per the copyright notice at the top of this file, I don't think we are allowed to include this as part of this repo.

iangrooms reviewed (changes requested):
Just one more change: please remove the `mkl_dfti.f90` file. After that I'll be ready to merge.

rachmeister commented:
@iangrooms so sorry for the oversight on that. I'll submit a new pull request on a clean branch with the current repo but omitting the file to keep the commits clean as well.
